lib/storage/utils.js

```diff
@@ -61,3 +61,39 @@ module.exports.setDirSyncFlag = function setDirSyncFlag(path, logger) {
         loggedWarning = true;
     }
 };
+
+// custom serializer function for JSON stringify to avoid circular JSON error
+// this code is a modified version of json-stringify-safe npm package found at
+// https://github.com/moll/json-stringify-safe/blob/master/stringify.js
+module.exports.serializeCircJSON = function serializeCircJSON() {
```

Comment:
One more thing I noticed is that you have multiple module.exports floating in the file. This should have been caught by the linter but I guess not.
```js
function foo() {
 ........
}

function bar() {
.......
}
module.exports = {
foo, bar
}
```
This is the pattern we should be using for exports.

Reply:
Fixed